platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java

```diff
@@ -399,6 +403,69 @@ public SecureCxfClientFactoryImpl(
     this.oauthSecurity = oauthSecurity;
   }
 
+  /**
+   * Constructs a factory that will return security-aware cxf clients. Once constructed, use the
+   * getClient* methods to retrieve a fresh client with the same configuration. Providing {@link
+   * WebClient} to interfaceClass will create a generic web client.
+   *
+   * <p>This factory can and should be cached. The clients it constructs should not be.
+   *
+   * @param endpointUrl the remote url to connect to
+   * @param interfaceClass an interface representing the resource at the remote url
+   * @param providers optional list of providers to further configure the client
+   * @param interceptor optional message interceptor for the client
+   * @param disableCnCheck disable ssl check for common name / host name match
+   * @param allowRedirects allow this client to follow redirects
+   * @param connectionTimeout timeout for the connection
+   * @param receiveTimeout timeout for receiving responses
+   * @param sourceId the id of the source
+   * @param discoveryUrl the oauth provider's discovery url
+   * @param clientId the client id registered with the oauth provider
+   * @param clientSecret the client secret registered with the oauth provider
+   * @param username the oauth flow to use
+   * @param password the oauth flow to use
+   * @param additionalOauthParameters the oauth flow to use
+   * @param oauthSecurity class used to set oauth tokens on clients
+   */
+  @SuppressWarnings("squid:S00107")
```

Comment:
We probably need to make a builder for this thing.

Reply:
Agreed it's getting too big. I believe there was another PR on master to add one more constructor.